classes/pointmarking_form.php

```diff
@@ -0,0 +1,139 @@
+
+        $mform->addElement('static', 'instructions', get_string('gradinginstructions', 'observation'),
+            format_text($instext, $insformat));
+
+        switch($prefill['res_type']) {
+            // Text input type.
+            case 0:
+                $mform->addElement('textarea', 'response', get_string('textinputtype', 'observation'),
+                    ['rows' => 3, 'cols' => 100]);
+                $mform->setType('response', PARAM_RAW);
+                $mform->addRule('response', get_string('required', 'observation'), 'required', null, 'client');
+        }
+
+        $mform->addElement('static', 'max_grade_display', get_string('maxgrade', 'observation'), $prefill['max_grade']);
+
+        $mform->addElement('text', 'grade_given', get_string('gradegiven', 'observation'));
+        $mform->setType('grade_given', PARAM_INT);
+        $mform->addRule('grade_given', get_string('err_numeric', 'form'), 'numeric', null, 'client');
+        $mform->addRule('grade_given', get_string('required', 'observation'), 'required', null, 'client');
+        $mform->addRule('grade_given', get_string('intgreaterthanorzero', 'observation'), 'regex', '/^[0-9]\d*$/', 'client');
+
+        // Extra comment block.
+        $mform->addElement('textarea', 'ex_comment', get_string('extracomment', 'observation'), ['rows' => 3, 'cols' => 100]);
+        $mform->setType('ex_comment', PARAM_RAW);
+
+        // Action buttons.
+        $buttonarray = [];
+        $buttonarray[] = $mform->createElement('submit', 'save', get_string('save', 'observation'));
+        $buttonarray[] = $mform->createElement('submit', 'saveandnext', get_string('saveandnext', 'observation'));
+        $mform->addGroup($buttonarray, 'buttonar', '', ' ', false);
+
+        // Submit Observation.
+        $mform->addElement('header', 'submitheader', get_string('submitobservation', 'observation'));
+
+        $mform->registerNoSubmitButton('submitobservation');
+        $mform->addElement('submit', 'submitobservation', get_string('submitobservation', 'observation'));
+
+        // Cancel buttons.
+        $mform->addElement('header', 'cancelheader', get_string('abandonobservation', 'observation'));
+
+        $mform->registerNoSubmitButton('abandonbutton');
+        $mform->addElement('submit', 'abandonbutton', get_string('abandonobservation', 'observation'));
+
+        // Hidden form elements.
+
+        // Note this has to remove the underscore or else the required_param check fails.
+        $mform->addElement('hidden', 'sessionid', $prefill['session_id']);
+        $mform->setType('sessionid', PARAM_INT);
+
+        $mform->addElement('hidden', 'pointid', $prefill['point_id']);
+        $mform->setType('pointid', PARAM_INT);
+
+        $mform->addElement('hidden', 'id', $prefill['point_id']);
+        $mform->setType('id', PARAM_INT);
+
+        $mform->addElement('hidden', 'max_grade');
+        $mform->setType('max_grade', PARAM_INT);
+
+        // Enforce validations.
```

Comment:
This shouldnt be needed, validation is always required on the server side if the calls are correct